aptos-move/framework/aptos-framework/sources/fungible_asset/fungible_store.move

```diff
@@ -0,0 +1,314 @@
+/// This defines a store of `FungibleAssetWallet` under each account.
+module aptos_framework::fungible_store {
+    use aptos_framework::create_signer;
+    use aptos_framework::fungible_asset::{Self, FungibleAssetWallet, FungibleAsset, FungibleAssetMetadata, TransferRef, metadata_from_wallet, BurnRef};
+    use aptos_framework::object::{Self, Object};
+    use aptos_std::smart_table::{Self, SmartTable};
+    use std::error;
+    use std::option::{Self, Option};
+    use std::signer;
+    #[test_only]
+    use aptos_framework::fungible_asset::verify;
+
+    /// The fungible asset wallet object existence error.
+    const EFUNGIBLE_ASSET_WALLET_OBJECT: u64 = 1;
+
+    /// Represents all the fungible asset wallet objects of an onwer keyed by the base metadata objects.
+    struct FungibleAssetStore has key {
+        index: SmartTable<Object<FungibleAssetMetadata>, Object<FungibleAssetWallet>>
+    }
+
+    /// Check the balance of an account.
+    public fun balance<T: key>(
+        account: address,
+        metadata: &Object<T>
+    ): u64 acquires FungibleAssetStore {
+        let metadata = fungible_asset::verify(metadata);
+        let afa_opt = get_account_fungible_asset_object(
+            account,
+            &metadata,
+            false
+        );
+        if (option::is_none(&afa_opt)) {
+            return 0
+        };
+        let wallet = option::destroy_some(afa_opt);
+        fungible_asset::balance(&wallet)
+    }
+
+    /// Return true if `account` allows ungated transfer.
+    public fun ungated_transfer_allowed<T: key>(
+        account: address,
+        metadata: &Object<T>
+    ): bool acquires FungibleAssetStore {
+        let metadata = fungible_asset::verify(metadata);
+        let afa_opt = get_account_fungible_asset_object(
+            account,
+            &metadata,
+            false
+        );
+        if (option::is_none(&afa_opt)) {
+            return true
+        };
+        let wallet = option::destroy_some(afa_opt);
+        fungible_asset::ungated_transfer_allowed(&wallet)
+    }
+
+    /// Enable/disable the direct transfer.
+    public fun set_ungated_transfer(
+        ref: &TransferRef,
+        account: address,
+        allow: bool
+    ) acquires FungibleAssetStore {
+        let metadata = fungible_asset::verify(&fungible_asset::transfer_ref_metadata(ref));
+        let afa_opt = get_account_fungible_asset_object(account, &metadata, !allow);
+        if (option::is_none(&afa_opt)) {
+            return
+        };
+        let wallet = option::destroy_some(afa_opt);
+        fungible_asset::set_ungated_transfer(ref, &wallet, allow);
+        maybe_delete(wallet);
+    }
```

Comment:
I think for this store, we should just expose the most trivial operations, let's consider deleting these

Reply:
If we delete this, users have to implement all those by themselves?